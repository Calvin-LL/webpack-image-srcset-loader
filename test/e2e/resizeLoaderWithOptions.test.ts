import path from "path";

import WISLWebpackTestCompiler from "./helpers/WISLWebpackTestCompiler";

describe.each([4, 5] as const)(
  "v%d resize loader with options",
  (webpackVersion) => {
    it("should work with options in webpackConfig", async () => {
      const compiler = new WISLWebpackTestCompiler({ webpackVersion });
      const bundle = await compiler.compile({
        loaderOptions: {
          sizes: ["300w", "original"],
        },
        resizeLoaderOptions: {
          quality: 80,
        },
      });

      expect(bundle.execute("main.js")).toMatchSnapshot("result");
    });

    it("should work with options in query", async () => {
      const loaderPath = path.resolve(__dirname, "../../test-dist/cjs.js");
      const loaderOptions = {
        sizes: ["300w", "original"],
      };
      const loaderOptionsString = JSON.stringify(loaderOptions);
      const resizeLoaderOptions = {
        quality: 80,
      };
      const resizeLoaderOptionssString = JSON.stringify(resizeLoaderOptions);

      const compiler = new WISLWebpackTestCompiler({ webpackVersion });
      const bundle = await compiler.compile({
        fileContentOverride: `__export__ = require('-!${loaderPath}?${loaderOptionsString}!webpack-image-resize-loader?${resizeLoaderOptionssString}!./Macaca_nigra_self-portrait_large.jpg');`,
      });

      expect(bundle.execute("main.js")).toMatchSnapshot("result");
    });
  }
);
