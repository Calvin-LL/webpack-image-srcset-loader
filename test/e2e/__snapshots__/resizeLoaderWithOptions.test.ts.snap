// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v4 resize loader with options should work with options in query: result 1`] = `
Object {
  "default": "76ab12b37821a8c18371948d1be1b6c1.jpg 300w, bd2b9fdf768b5eb139702c4c3b4573de.jpg",
}
`;

exports[`v4 resize loader with options should work with options in webpackConfig: result 1`] = `"76ab12b37821a8c18371948d1be1b6c1.jpg 300w, bd2b9fdf768b5eb139702c4c3b4573de.jpg"`;

exports[`v5 resize loader with options should work with options in query: result 1`] = `
Object {
  "default": "76ab12b37821a8c18371948d1be1b6c1.jpg 300w, bd2b9fdf768b5eb139702c4c3b4573de.jpg",
}
`;

exports[`v5 resize loader with options should work with options in webpackConfig: result 1`] = `"76ab12b37821a8c18371948d1be1b6c1.jpg 300w, bd2b9fdf768b5eb139702c4c3b4573de.jpg"`;
